meerk40t/gui/tips.py

```diff
@@ -167,84 +169,103 @@ def current_tip(self, newvalue):
             self.set_tip_image(my_tip[2], newvalue, self.context.tip_access_consent)
         else:
             self.set_tip_image("", newvalue, self.context.tip_access_consent)
+        self.label_position.SetLabel(_("Tip {idx}/{maxidx}").format(idx=newvalue + 1, maxidx=len(self.tips)))
         self.Layout()
 
-    def set_tip_image(self, path, counter, automatic_download, display=True):
+    def _download_image(self, uri, local_file):
+        """
+        Download image from given URI to local_file.
+        @param uri:
+        @param local_file:
+        @return:
+        """
+        if not self.context.tip_access_consent:
+            raise PermissionError("We do not have consent to download.")
+        try:
+            with urllib.request.urlopen(uri) as file:
+                content = file.read()
+        except (urllib.error.URLError, urllib.error.HTTPError) as e:
+            # print (f"Error: {e}")
+            return False
+        # If the file object is successfully opened, read its content as a string
+        try:
+            with open(local_file, "wb") as f:
+                f.write(content)
+            return True
+        except (OSError, PermissionError, RuntimeError) as e:
+            # print (f"Error @ image write to {local_path}: {e}")
+            return False
+
+    def set_tip_image(self, path, counter, automatic_download):
         """
         path: URL of the image to be loaded
         counter: an additional naming element to
         avoid ending up with the same name for equally called images
         display: apply and display the image
         """
-        self.tip_image = path
+        self.no_image_message.Show(False)
```

Comment:
De-nesting led to a couple of cases where the previous ie invalid image was still displayed